### ISSUE TITLE
tweedie loss function

### ISSUE THREAD

**Issue by Alit10** [microsoft/LightGBM#6867]
Hello,
I tried to define a custom loss using tweedie. 
So I did a first test define grandient and hessian without modification and comparing to the run with objective = 'tweedie' and I have two different results 

Any ideas why ?

My code:
`    grad_tweedie = (-y_true * np.exp((1 - p) * y_pred) + np.exp((2 - p) * y_pred)) * w
    hess_tweedie = (-y_true * (1 - p) * np.exp((1 - p) * y_pred) + (2 - p) * np.exp((2 - p) * y_pred)) * w`
